gcovr/tests/test_gcov_parser.py

```diff
@@ -558,6 +564,68 @@ def test_branch_exclusion(flags):
     assert covered_branches == expected_covered_branches
 
 
+def test_negativ_branch_count():
+    """
+    A exception shall be raised.
+    """
+
+    source = textwrap.dedent(
+        """\
+          1: 1: normal line
+        branch 1 taken 80%
+          1: 2: } // line without apparent code
+        branch 2 taken -11234
+          1: 3: exception-only code
+        branch 3 taken 60% (throw)
+        """
+    )
+
+    try:
+        parse_coverage(
+            source.splitlines(),
+            filename="example.cpp",
+            ignore_parse_errors=False,
+            ignore_negative_counters=False,
+        )
+    except Exception as e:
+        assert isinstance(e, NegativeCounter)
+    else:
+        assert False, "Here we should never come"
```

Comment:
Could be simplified using `pytest.raises`:

```suggestion
    with pytest.raises(NegativeCounter):
        parse_coverage(
            source.splitlines(),
            filename="example.cpp",
            ignore_parse_errors=False,
            ignore_negative_counters=False,
        )
```

See: https://docs.pytest.org/en/7.2.x/how-to/assert.html#assertions-about-expected-exceptions